### PR TITLE
[RFR] Use a graphical board

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Use a graphical board

### CODE REVIEW

File: src/components/grid.js [hunk 1]
```diff
@@ -3,36 +3,46 @@ import PropTypes from 'prop-types';
...
-);
+import { associateTileToBackground } from '../core/helper';
+
+const Grid = ({ grid, onClick, resolvedGrid, readOnly }) => {
+    const url = '/static/images/sea.jpg';
+    const tileToBg = associateTileToBackground(resolvedGrid);
+
+    return (
+        <div className="puzzle-column">
+            {grid.map((row, rowKey) => {
```
Comment (jdemangeon) [inline src/components/grid.js]: Use implicit return () => ({...})

File: src/core/helper.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const associateTileToBackground = grid => {
+    const tileSize = 100 / grid.length; // 100%
+    let tileToBg = {};
+    grid.forEach((row, rowKey) => {
```
Comment (jdemangeon) [inline src/core/helper.js]: Prefer reduce use

File: src/components/grid.js [hunk 3]
```diff
@@ -3,36 +3,46 @@ import PropTypes from 'prop-types';
...
-                    )}
-                </div>
-            );
-        })}
-    </div>
-);
+import { associateTileToBackground } from '../core/helper';
+
+const Grid = ({ grid, onClick, resolvedGrid, readOnly }) => {
+    const url = '/static/images/sea.jpg';
```
Comment (Kmaschta) [inline src/components/grid.js]: If its a constant, you can move this constant out of the Grid.
This way, you won't instanciate `url` in memory for each Grid.

File: src/core/helper.js (hunk 2 above)
Comment (Luwangel) [reply src/core/helper.js]: I find that use two reduce is less readable.

File: pages/game.js [hunk 4]
```diff
@@ -76,11 +82,15 @@ export default class Index extends Component {
...
-                        />
+                        {currentGrid && (
+                            <Grid
+                                onClick={this.handleClick}
+                                grid={currentGrid}
+                                resolvedGrid={resolvedGrid}
+                                readOnly={isVictory}
+                            />
+                        )}
+                        {!currentGrid && <div />}
```
Comment (jdemangeon) [inline pages/game.js]: This empty div is temporary?

File: src/components/tile.js [hunk 5]
```diff
@@ -1,20 +1,30 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import ClassNames from 'classnames';
 
-const Tile = ({ enabled, onClick, tileValue }) => {
+const Tile = ({ enabled, onClick, tileImage, tileImageCoords, tileValue }) => {
     const handleClick = () => {
```
Comment (jdemangeon) [inline src/components/tile.js]: You should not re-declare function at each render.  Create a "class" or use "[withProps](https://github.com/acdlite/recompose/blob/master/docs/API.md#withprops)"  in the following way instead.

```
return withProps(({ enabled, onClick, tileValue }) => ({
  handleClick: () => enabled && onClick(tileValue),
}))(Tile);
```

**Review by jdemangeon — APPROVED** [review]

File: src/core/helper.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const associateTileToBackground = grid => {
```
Comment (floo51) [inline src/core/helper.js:1]: This  is not tested

**Review by floo51 — CHANGES_REQUESTED** [review]

**Review by floo51 — APPROVED** [review]

**Pull request merged**